modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -108,6 +109,7 @@ public JavaClientCodegen() {
 
         featureSet = getFeatureSet().modify()
                 .includeDocumentationFeatures(DocumentationFeature.Readme)
+                .includeGlobalFeatures(GlobalFeature.ParameterizedServer)
```

Comment:
AFAICS this was only implemented for Jersey2 client in https://github.com/OpenAPITools/openapi-generator/pull/4998 - can we mark this feature as included based on the API library used when generating?

Reply:
Great call out. I added this here because the ServerConfiguration and ServerVariable types were added to all generators. I wonder if that was intentional and those files should only be added if Jersey is selected.

I can add this to the libraries option. We're currently not outputting library specific feature matrix.

Reply:
IIUC it would be better to only add these files when jersey2 is used, CC @jirikuncar who authored the PR to confirm.

Reply:
Yes, the parameterised server per endpoint support is only in jersey2.

Reply:
Hm. I'm glad this happened because it made me realize that I screwed up the location of the `featureSet` property. I'd designed it to be a property on `GeneratorMetadata` alongside `getLibraryFeatures`, but for some reason I made the property on `GeneratorMetadata` _and_ `DefaultCodegen`.

I'll leave this option where it is now on this PR, add a TODO and move the `supportingFiles` additions to only occur when jersey is selected… then I'll open another PR to move `featureSets` to `generatorMetadata` and include the library-specific option in that PR.

Reply:
There is a support for global server URL **generation** in all Java clients, that can be used instead of `basePath`. Additionally Jersey2 has support for server URL configuration per endpoint.

Reply:
I was not clear. There are actually 2 features to track:

1. Support for global server configuration with templated values (all Java)
2. Support for server configuration per endpoint (only jersey2, go-experimental)

Reply:
So, we probably need a `GlobalFeature.ParameterizedServerPerOperation`?

Reply:
I created #5402 to track adding another feature for per-operation server variables.